dpkg-buildpackage support

hellermf opened the pull request:
Several intertwined changes:
- Added a top level ./debian folder for packaging a monolithic LStore package
- Added support for versioning the shared libraries
- Added pkgconfig metadata
- Added support for -dbgsym (aka ddeb) package on Debian/Ubuntu/...
- Added support for -dev package on Debian/Ubuntu/...
  .
  Notes:
- kept the *.ex sample templates in ./debian, fine if the maintainer wants to rm
- This supports dpkg-buildpackage calling cmake,
  this does not use the CPack stuff in LStorePackaging.cmake
- Run "dpkg-buildpackage -us -uc" in the project folder and debs should appear in
  the projects parent folder after a few minutes of building

hellermf committed:
dpkg-buildpackage support

Several intertwined changes:
* Added a top level ./debian folder for packaging a monolithic LStore package
* Added support for versioning the shared libraries
* Added pkgconfig metadata
* Added support for -dbgsym (aka ddeb) package on Debian/Ubuntu/...
* Added support for -dev package on Debian/Ubuntu/...
.
Notes:
- kept the *.ex sample templates in ./debian, fine if the maintainer wants to rm
- This supports dpkg-buildpackage calling cmake,
  this does not use the CPack stuff in LStorePackaging.cmake
- Run "dpkg-buildpackage -us -uc" in the project folder and debs should appear in
  the projects parent folder after a few minutes of building

hellermf committed:
dpkg-buildpackage support

Several intertwined changes:
* Added a top level ./debian folder for packaging a monolithic LStore package
* Added support for versioning the shared libraries
* Added pkgconfig metadata
* Added support for -dbgsym (aka ddeb) package on Debian/Ubuntu/...
* Added support for -dev package on Debian/Ubuntu/...
.
Notes:
- kept the *.ex sample templates in ./debian, fine if the maintainer wants to rm
- This supports dpkg-buildpackage calling cmake,
  this does not use the CPack stuff in LStorePackaging.cmake
- Run "dpkg-buildpackage -us -uc" in the project folder and debs should appear in
  the projects parent folder after a few minutes of building

hellermf committed:
dpkg-buildpackage support

Several intertwined changes:
* Added a top level ./debian folder for packaging a monolithic LStore package
* Added support for versioning the shared libraries
* Added pkgconfig metadata
* Added support for -dbgsym (aka ddeb) package on Debian/Ubuntu/...
* Added support for -dev package on Debian/Ubuntu/...
.
Notes:
- kept the *.ex sample templates in ./debian, fine if the maintainer wants to rm
- This supports dpkg-buildpackage calling cmake,
  this does not use the CPack stuff in LStorePackaging.cmake
- Run "dpkg-buildpackage -us -uc" in the project folder and debs should appear in
  the projects parent folder after a few minutes of building

hellermf committed:
dpkg-buildpackage support

Several intertwined changes:
* Added a top level ./debian folder for packaging a monolithic LStore package
* Added support for versioning the shared libraries
* Added pkgconfig metadata
* Added support for -dbgsym (aka ddeb) package on Debian/Ubuntu/...
* Added support for -dev package on Debian/Ubuntu/...
.
Notes:
- kept the *.ex sample templates in ./debian, fine if the maintainer wants to rm
- This supports dpkg-buildpackage calling cmake,
  this does not use the CPack stuff in LStorePackaging.cmake
- Run "dpkg-buildpackage -us -uc" in the project folder and debs should appear in
  the projects parent folder after a few minutes of building

PerilousApricot commented:
Sick. I'll merge this in shortly

## 

It's dark in this basement.

PerilousApricot commented on CMakeLists.txt:
```diff
@@ -5,6 +5,10 @@ if(NOT CMAKE_BUILD_TYPE)
 
+if(NOT LSTORE_PROJECT_VERSION)
+    set(LSTORE_PROJECT_VERSION "0.5.1-unknown_commit")
+endif()
```
This can be set like the following below:

```
set(LSTORE_REPO_URL "http://repo.accre.vanderbilt.edu"
         CACHE STRING "URL for lstore-release")
```

PerilousApricot commented:
Everything looks good except for the one comment above. Sphinx generates man pages from the same sources, so once that hits & gets into the build chain, I'll drop the in-built ones in favor of autogenerating from sphinx. Once that happens, is it ~easy to have "dynamic" man pages added to the package? I'm woefully unfamiliar with dpkg (need to learn it soon, though!)

PerilousApricot commented:
Fixes #49

hellermf committed:
Improved style of set-default-value constructs in CMake

hellermf committed:
Improved style of set-default-value constructs in CMake

hellermf committed:
Improved style of set-default-value constructs in CMake

hellermf commented on CMakeLists.txt:
```diff
@@ -1,13 +1,10 @@
-if(NOT LSTORE_PROJECT_VERSION)
-    set(LSTORE_PROJECT_VERSION "0.5.1-unknown_commit")
-endif()
+set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Build type (Debug|Release)")
```
Note that with the addition of -dbgsym package creation the meaning of CMAKE_BUILD_TYPE gets muddled. In this scenario you want your compile flags to be more like the debug flags and then the binaries later get separated into stripped release binaries and debug symbol files. This should be addressed somehow.

PerilousApricot replied:
Good point. There's a ReleaseWithDebugInfo build type (I think that's the
name....maybe RelWithDebInfo)

## 

It's dark in this basement.

hellermf committed:
Update Debian/CZMQ packaging (#40)

Get Debian and CZMQ to build properly

PerilousApricot committed:
First pass at an API documentation skeleton (#51)

Begin getting framework together to generate sphinx + doxygen + breathe
documentation for LStore. First document toolbox/chksum.h to produce a
sample that can later be expanded

PerilousApricot commented:
Alright, mind rebasing on top of master again?

hellermf committed:
Merge 'feature/update-packaging-matt' into feature/update-packaging-matt

PerilousApricot committed:
Change visibility of libraries (#52)

Instead of exporting everything by default into the shared libraries,
hide everything by default and only export things explicitly.

hellermf committed:
Merge hellermf/feature/update-packaging-matt & feature/update-packaging-matt

hellermf committed:
Bogus commit to trigger Jenkins build

PerilousApricot commented:
Committed manually with 368e45959a88f783bc48a3464762cf92f773d71e
